packages/core/src/validate.js

```diff
@@ -5,6 +5,7 @@ import { deepEquals, getDefaultFormState } from "./utils";
 
 let formerCustomFormats = null;
 let formerMetaSchema = null;
+const ROOT_SCHEMA_PREFIX = "__rjsf_rootSchema";
```

Comment:
Also, just double-checking -- are we positive that all refs that are currently recognized by rjsf and/or valid according to the JSON Schema spec will start with `#`? If not, we might be breaking some functionality.

Reply:
not sure what you're concern is exactly. Do you want to check that ref starts with `#` before adding the prefix?

Reply:
Looking at this again, I don't think people will ever have ref's that don't start with `#` (because rjsf crashes otherwise!) So we should be okay with just the existing code.

Reply:
I actually went ahead and added 

```javascript
if (
        key === "$ref" &&
        typeof value === "string" &&
        value.startsWith("#")
      )
...
```

I assume its fine having that check anyways